packages/react-components/components/CallToActionsBar.tsx

```diff
@@ -15,16 +15,25 @@ interface Props {
 export default function CallToActionsBar({ callToActions, testID }: Props) {
   return (
     <View style={styles.container} testID={testID}>
-      {callToActions.map((cta, i) => (
-        <TextButton
-          testID={`${testID}/${cta.text}/Button`}
-          key={i}
-          style={styles.action}
-          onPress={cta.onPress}
-        >
-          {cta.text}
-        </TextButton>
-      ))}
+      {callToActions.map((cta, i) => {
+        if (typeof cta.text === 'string') {
+          return (
+            <TextButton
+              testID={`${testID}/${cta.text}/Button`}
+              key={i}
+              style={styles.action}
+              onPress={cta.onPress}
+            >
+              {cta.text}
+            </TextButton>
+          )
+        }
+        return (
+          <View key={i} style={styles.action}>
```

Comment:
this doesnt need to have onPress? or styles?

Reply:
Nope it's a JSX element (activity spinner)